### PR TITLE
Feature/notifications

### PR DESCRIPTION
**Pull request by Feld0**
This is the backend piece of #25.

### CODE REVIEW

File: app/Library/Notifications/NotificationManager.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * This class exists mostly to maintain type safety when sending notifications
+ * from around the Pony.fm codebase. There should be virtually zero logic here.
+ * All the heavy lifting happens asynchronously in the {@link SendNotifications}
+ * job and the notification drivers.
+ */
+class NotificationManager implements NotificationHandler {
+    use DispatchesJobs;
+
+    private function dispatchNotification(string $notificationType, array $notificationData) {
+        $this->dispatch( (new SendNotifications($notificationType, $notificationData))->onQueue('notifications') );
```
Comment (Feld0) [inline app/Library/Notifications/NotificationManager.php]: I already configured this worker queue in Ansible, FYI.

**Comment by logic-dev** [comment]
Performing any action that sends a notification returns an error, the action completes but the notification does not send

```
FatalThrowableError in PonyfmDriver.php line 41:
Type error: Argument 2 passed to Poniverse\Ponyfm\Library\Notifications\Drivers\PonyfmDriver::insertNotifications() must implement interface ArrayAccess, array given, called in /vagrant/app/Library/Notifications/Drivers/PonyfmDriver.php on line 106
```

**Comment by Feld0** [comment]
@logic-dev Thanks for catching that - I pushed a fix.

File: app/Jobs/SendNotifications.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $this->beforeHandle();
+        
+        // get list of users who are supposed to receive this notification
+        $recipients = [User::find(1)];
```
Comment (Lavoaster) [inline app/Jobs/SendNotifications.php]: Is this still supposed to be hardcoded?

Comment (Feld0) [inline app/Jobs/SendNotifications.php]: nopenopenope. Thanks for catching this. The right end result is achieved, incidentally, but it should not happen like this. 😱

Comment (logic-dev) [inline app/Jobs/SendNotifications.php]: 2ofuba1

**Pull request merged**